bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/UpnpDiscoveryService.java

```diff
@@ -102,6 +104,7 @@ protected void startScan() {
         }
         upnpService.getRegistry().addListener(this);
         upnpService.getControlPoint().search();
+        upnpService.getControlPoint().search(new RootDeviceHeader());
```

Comment:
Yes, this could duplicate the amount of UPnP messages in the network, cause all RootDevices which answered to the original getControlPoint().search() request will answer again.
Until a UPnP guru comes up with a way to do a "device specific" request like what I used in my own SSDP discovery (using the STHeader with "ST: urn:Belkin:service:basicevent:1"), we will have to live with that.

Background discovery is also working, so plugging a new WeMo Insight into the socket while smarthome is running, it will announce itselve and be put into inbox. I just verified this again.